src/common/valnum.cpp

```diff
@@ -234,9 +234,29 @@ wxString wxIntegerValidatorBase::ToString(LongestValueType value) const
 }
```

Comment:
`wxIntegerValidatorBase::ToString` should be updated too:

```
{
    if ( CanBeNegative() )
        return wxNumberFormatter::ToString(value, GetFormatFlags());

    wxULongLong_t uvalue = static_cast<wxULongLong_t>(value);
    return wxNumberFormatter::ToString(uvalue, GetFormatFlags());
}
```
Otherwise `wxNumValidatorBase::OnKillFocus` would leave the control with a negative value after normalizing its content.

Reply:
Hmm, how exactly is this going to happen? I'd like to add a unit test for this, but I'm not sure what exactly needs to be done to trigger this, do you know?

Reply:
The problem is at this [line](https://github.com/wxWidgets/wxWidgets/blob/master/include/wx/valnum.h#L257),
i.e. `s = this->ToString(value);`

`OnKillFocus` calls `NormalizeString` which calls `NormalizeValue` thinking they are dealing with a signed value! (i.e. `LongestValueType`).

Reproduced by this:
```
TEST_CASE_METHOD(NumValidatorTestCase, "ValNum::TransferULL", "[valnum]")
{
...
    CHECK( valULL.TransferToWindow() );
    CHECK( m_text->GetValue() == "18446744073709551615" ); // <-- fails!
}
```

Reply:
Yes, you're absolutely right, sorry for not realizing it sooner. Fixed now, with more tests to check for it, thanks again!